rust/settings/BUILD.bazel

```diff
@@ -9,6 +10,26 @@ incompatible_flag(
     issue = "https://github.com/bazelbuild/rules_rust/issues/966",
 )
 
+# A flag controlling whether to rename first-party crates such that their names
+# encode the Bazel package and target name, instead of just the target name.
+#
+# First-party vs. third-party crates are identified using the value of
+# //settings:third_party_dir.
+incompatible_flag(
```

Comment:
I assumed we will maintain both naming strategies for some time, so using an `incompatible_flag` for this right away may send a wrong impression to users (typically users have to migrate for `incompatible_flags` rather promptly). Would it make sense to you if we made this a bool_flag for the time being, and only added incompatible flag after long time and proper discussion with other owners?

That said, if we set `third_party_dir` to `//` by default, and we made sure it works for external repositories, we could enable the behavior of `rename_1p_crates` even without the incompatible flag today, it's just that most (all? except us?) users would still be using the third-party like behavior for all their workspaces. Let's not do this in this PR though.

Reply:
Thanks, I wasn't sure if `incompatible_flag` was appropriate for this, but in hindsight that should have been clear. I didn't mean to propose this as "the only option, in the future", but as an option that will be supported. I'll make it a `bool_flag`, and rename it to `rename_first_party_crates` per https://github.com/bazelbuild/rules_rust/issues/1013#issuecomment-986900290, while I'm at it.

It's an interesting thought to make this the default/only behavior, and rely on people to use the third-party opt-out. But I agree, I think that kind of change should not be part of this PR.

Reply:
Agreed on all points.